src/core/components/hv-navigator/types.ts

```diff
@@ -32,5 +33,5 @@ export type NavigatorParams = {
  */
 export type Props = {
   element: TypesLegacy.Element;
-  routeComponent: FC;
+  routeComponent: FC<HvRouteProps>;
```

Comment:
Should we create a local version of the props rather than reaching across modules?

```
/**
 * All of the props used by hv-route
 */
export type RouteProps = {
  navigation?: NavigatorService.NavigationProp;
  route?: NavigatorService.Route<string, RouteParams>;
};
```

Reply:
is this desirable? I would imaging Props type of a component to be part of the public interface of a module, hence available for other modules to consume. Are there down sides in your opinion in this approach?

Reply:
Just keeping modules insular.